<h1><b>'likes' Data Report</b></h1>
<h2>Jacob Nicholson</h2>
<h3>8/13/24</h3>

<p>The data for this assignment is collected from my perosnal instagram account. An entry was created for everytime I have cicked to "like" a post by other users, it can be used to understand my connections to other user's, and my general intrests. This data is somewhat reliable because it is only me who used this account and comes directly from the source. However there is a chance posts that are liked could have since been deleted, or accidentally clicked.</p>
<hr>

In [164]:
import json as js
import pandas as pd

In [165]:
path = "/Users/jacobnicholson/DataInEmergingMediaAndTech/instagram-jl.nicholson-2024-08-27-HpSzDfYz/your_instagram_activity/likes/liked_posts.json"
with open(path) as j:
    likes = js.load(j)

In [166]:
likes_df = pd.DataFrame(likes['likes_media_likes'])
likes_df.head()

title                                   string_list_data
0  ifyouhigh  [{'href': 'https://www.instagram.com/reel/CSxP...
1  ifyouhigh  [{'href': 'https://www.instagram.com/reel/CTNb...
2  ifyouhigh  [{'href': 'https://www.instagram.com/reel/CTkO...
3  ifyouhigh  [{'href': 'https://www.instagram.com/reel/CTkn...
4  ifyouhigh  [{'href': 'https://www.instagram.com/reel/CTr9...

In [167]:
likes_df.groupby('title').count().\
sort_values('string_list_data', ascending = False)

string_list_data
title                                   
practicingselfabuse                  145
fanmaps                               93
hard.images                           73
mapsareweapons                        65
bluereviewprojectspace                58
...                                  ...
hayesmajerus                           1
hayk_animation                         1
hayvanormal                            1
haywhy_dc                              1
zzzoppka                               1

[10695 rows x 1 columns]

In [168]:
likes_df2 = pd.DataFrame([x[0] for x in likes_df['string_list_data']])

In [169]:
likes_df2['title'] = likes_df['title']
likes_df2.head()

href value   timestamp      title
0  https://www.instagram.com/reel/CSxP9pLn9kH/  👍  1633794817  ifyouhigh
1  https://www.instagram.com/reel/CTNb-fbn79F/  👍  1633794738  ifyouhigh
2  https://www.instagram.com/reel/CTkOnlpnIKm/  👍  1633794626  ifyouhigh
3  https://www.instagram.com/reel/CTknXyMH6dR/  👍  1633794608  ifyouhigh
4  https://www.instagram.com/reel/CTr9U5djOpf/  👍  1633794573  ifyouhigh

In [170]:
likes_df2.groupby('title').count().sort_values('href', ascending = False).head(10)

href  value  timestamp
title                                         
practicingselfabuse      145    145        145
fanmaps                   93     93         93
hard.images               73     73         73
mapsareweapons            65     65         65
bluereviewprojectspace    58     58         58
tbftcb                    57     57         57
falseknees                54     54         54
honeynut.woomy            52     52         52
ape_archive               42     42         42
sophsalv                  41     41         41

In [171]:
likes_df3 = likes_df2.drop(columns = ['href'])

In [172]:
likes_df3

value   timestamp                   title
0      👍  1633794817               ifyouhigh
1      👍  1633794738               ifyouhigh
2      👍  1633794626               ifyouhigh
3      👍  1633794608               ifyouhigh
4      👍  1633794573               ifyouhigh
...     ...         ...                     ...
19412  👍  1679375310  trendyandthemultiverse
19413  👍  1679375103  trendyandthemultiverse
19414  👍  1679375004               filmoment
19415  👍  1679374118             stinkyasher
19416  👍  1679369445            akamztweenty

[19417 rows x 3 columns]

In [173]:
likes_df3.groupby('title').count().sort_values('value', ascending = False).head(10)

value  timestamp
title                                   
practicingselfabuse       145        145
fanmaps                    93         93
hard.images                73         73
mapsareweapons             65         65
bluereviewprojectspace     58         58
tbftcb                     57         57
falseknees                 54         54
honeynut.woomy             52         52
ape_archive                42         42
sophsalv                   41         41

<h2><b>Hypothesis:</b></h2>
<P>I am more likely to like multiple posts from accounts I follow back. Meaning out of every account with two or more likes from me, 51% of them follow me back. To test this I will use the "followers_1.json", I belive this should contain every account who is following me which I should be able to compare with my liked dataframe.</P>

<h6><i>Also please ignore account names, I think it many are meme pages I use to follow many years ago</i></h6>

In [174]:
import json as js
import pandas as pd
path = "/Users/jacobnicholson/DataInEmergingMediaAndTech/instagram-jl.nicholson-2024-08-27-HpSzDfYz/connections/followers_and_following/followers_1.json"
with open(path) as a:
    followers = js.load(a)

In [175]:
followers_df = pd.DataFrame(followers)
followers_df.head()

title media_list_data                                   string_list_data
0                    []  [{'href': 'https://www.instagram.com/jackseipl...
1                    []  [{'href': 'https://www.instagram.com/austinsf0...
2                    []  [{'href': 'https://www.instagram.com/ksurunnin...
3                    []  [{'href': 'https://www.instagram.com/elprupaar...
4                    []  [{'href': 'https://www.instagram.com/anushka_d...

In [176]:
followers_df2 = pd.DataFrame(followers).drop(columns = ['title', 'media_list_data'])
followers_df2.head()

string_list_data
0  [{'href': 'https://www.instagram.com/jackseipl...
1  [{'href': 'https://www.instagram.com/austinsf0...
2  [{'href': 'https://www.instagram.com/ksurunnin...
3  [{'href': 'https://www.instagram.com/elprupaar...
4  [{'href': 'https://www.instagram.com/anushka_d...

In [177]:
followers_df3 = pd.DataFrame([x[0] for x in followers_df2['string_list_data']])
followers_df3.head()

href          value   timestamp
0     https://www.instagram.com/jackseiple     jackseiple  1724443384
1     https://www.instagram.com/austinsf02     austinsf02  1724166211
2     https://www.instagram.com/ksurunning     ksurunning  1724113647
3   https://www.instagram.com/elprupaaronl   elprupaaronl  1723728578
4  https://www.instagram.com/anushka_dhage  anushka_dhage  1723728568

In [178]:
followers_df4= pd.DataFrame(followers_df3).drop(columns = ['href'])
followers_df4.head()

value   timestamp
0     jackseiple  1724443384
1     austinsf02  1724166211
2     ksurunning  1724113647
3   elprupaaronl  1723728578
4  anushka_dhage  1723728568

In [179]:
followers_df4.groupby('value').count().sort_values('timestamp', ascending = False).head(100)

timestamp
value                     
03c23c04                 1
luuisamachado            1
marissa.schmitt          1
marinanne2005            1
malloryt03               1
...                    ...
lancaster_pal            1
kyle.garber              1
kylayetter.sd            1
ksurunning               1
konnor.83                1

[100 rows x 1 columns]

<h4>In order to help me compare these two tidied data frames, the next lines were written with the help of ChatGPT4</h4>

In [180]:
common_index = likes_df3[likes_df3['title'].isin(followers_df4['value'])]
print("Common titles in likes_df3 that are also in followers_df4:", common_index)

Common titles in likes_df3 that are also in followers_df4:       value   timestamp             title
82     👍  1633356401    alex_mobeer612
100    👍  1633233913  benjaminschenck_
151    👍  1632794357     carlie.dozier
157    👍  1632687486    gavinnmitchell
158    👍  1632657982      brandon._.he
...     ...         ...               ...
19256  👍  1680116710      cm_sheehan65
19266  👍  1680034715            tbftcb
19275  👍  1679964582      ishankmaitra
19327  👍  1679706799      arthur_zablo
19344  👍  1679600895       joshua._.67

[1066 rows x 3 columns]


In [181]:
# Group by 'title', count the occurrences, and sort by 'value'
grouped_likes_df = likes_df3.groupby('title').count().sort_values('value', ascending=False)

# Filter to only include entries where the count of 'value' is 2 or more
filtered_likes_df = grouped_likes_df[grouped_likes_df['value'] >= 2]

# Display the top 2000 rows
top_2000_filtered_likes_df = filtered_likes_df.head(2000)

print(top_2000_filtered_likes_df)

                        value  timestamp
title                                   
practicingselfabuse       145        145
fanmaps                    93         93
hard.images                73         73
mapsareweapons             65         65
bluereviewprojectspace     58         58
...                       ...        ...
misharkx                    2          2
levetatete                  2          2
lettuce.scream              2          2
mistergash                  2          2
yves.paintkillers           2          2

[2000 rows x 2 columns]


<h5>Above is account filtered to show only those with two or more liked posts from me</h5>

In [182]:
common_index = filtered_likes_df.isin(followers_df4['value'])
print("Common titles in likes_df3 that are also in followers_df4:", common_index)

Common titles in likes_df3 that are also in followers_df4:                         value  timestamp
title                                   
practicingselfabuse     False      False
fanmaps                 False      False
hard.images             False      False
mapsareweapons          False      False
bluereviewprojectspace  False      False
...                       ...        ...
tonsser_uk              False      False
ayoedebiri              False      False
big.human               False      False
bigguyjonllc            False      False
cricken_                False      False

[2538 rows x 2 columns]


<h5>This now searches for the usersnames found in the followers file inside of the liked post file and prints wether that boolean is true or false for each entry in the dataframes</h5>

In [183]:
# Check which values in 'filtered_likes_df' are in 'followers_df4['value']'
common_index = filtered_likes_df.index.isin(followers_df4['value'])

# Create a new DataFrame with only the rows where common_index is True
common_titles_df = filtered_likes_df[common_index]

print("Filtered DataFrame with common titles in likes_df3 and followers_df4:")
print(common_titles_df)


Filtered DataFrame with common titles in likes_df3 and followers_df4:
                         value  timestamp
title                                    
tbftcb                      57         57
sophsalv                    41         41
bren_ev3                    33         33
kdizzledrizzle              31         31
cecilia._ellen              29         29
...                        ...        ...
jenna.sutt                   2          2
checan.davi                  2          2
calvin.halliday              2          2
christina.s.nicholson.5      2          2
bowiex_x                     2          2

[144 rows x 2 columns]


<h2>Conclusion</h2>
<p>This text print from the tidy data frames represent the amount of my followers who's posts I have liked atleast 2 or more of. When compared with the previous text print, one can infer my hypothesis to be false, as out of the total 2538 users I have liked 2 or more posts of, only 144 of them (17.625%) are users who follow me, which is far less than the 51% I expected. I do think that my two orignal dataframes are tidy, one containing only the username, ammount of likes, and timestamp, and another containg only the usernames and timestamp, altough perhaps neither truly need the timestamp column.</p>